### ISSUE TITLE
Provide onUpgrade example

### ISSUE THREAD

**Issue by Nik99** [tekartik/sqflite#88]
It would be nice to have an onUpgrade example to follow with some best practices for devs that are using sqlite and this library for the first time.

**Comment by alextekartik** [comment]
Agree, it is just the matter of taking the time for it so contributions are welcome, I could also add references to existing and new blog post or tutorials.

**Comment by Nik99** [comment]
For sure, when you have time or contributors do have!
Thanks, you are really nice.

**Comment by hhfa1990** [comment]
Yes please!

**Comment by alextekartik** [comment]
There is an example available here:

https://github.com/tekartik/sqflite/blob/master/sqflite/doc/migration_example.md

**Comment by mikeevstropov** [comment]
If i have 3 versions, will it call onUpgrade twice to upgrade from 1 to 2 and 2 to 3, or only once (1 to 3)? Thank you.

**Comment by alextekartik** [comment]
@mikeevstropov only once

**Comment by Dooks123** [comment]
I created the following method which I find easier to understand:

    Future<void> _onUpgrade(Database db, int oldVersion, int newVersion) async {
      for (var version = oldVersion + 1; version <= newVersion; version++) {
        switch (version) {
          case 1: {
            //Version 1 - no changes
            break;
          }
          case 2: {
            await db.execute('ALTER TABLE xxx ADD COLUMN yyy INTEGER DEFAULT 0');
            break;
          }
        }
      }
    }

If your database is currently on `version 3`, and you update to `version 5`, it will enter the case statement in `version 4` and `version 5`.
